### PR TITLE
Eliminates redundant array creation in convinience NSString.init()

### PR DESCRIPTION
**Pull request by valeriyvan**

### CODE REVIEW

File: Sources/Foundation/NSString.swift [hunk 1]
```diff
@@ -231,8 +231,8 @@ open class NSString : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSC
     }
     
     public override convenience init() {
-        let characters = Array<unichar>(repeating: 0, count: 1)
-        self.init(characters: characters, length: 0)
+        var nul: unichar = 0
+        self.init(characters: &nul, length: 0)
     }
```
Comment (spevans) [inline Sources/Foundation/NSString.swift:236]: Since this is ultimately an empy string, can the whole function body be reduced to 
```
self.init("")
```
?

Comment (valeriyvan) [reply Sources/Foundation/NSString.swift:236]: yes, it could.

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test and merge

**Pull request merged**